### ISSUE TITLE
Tflearn accuracy on a large input is always 1

### ISSUE THREAD

**Issue by sabersf** [tflearn/tflearn#669]
I'm training a regression DNN, and the output of a last layer should be a number between [-1,1]. It seems like the network is considering the output layer as categorical and always produces the output either -1 or 1 and nothing in between. The acc number is always 1 (Which is strange on a large input). 

Here's the network:
`with tf.Graph().as_default():
    tnorm = tflearn.initializations.uniform(minval=-1.0, maxval=1.0)
    net = tflearn.input_data(shape=[None, 128])
    net = tflearn.fully_connected(net, 64, activation='tanh', weights_init=tnorm)
    net = tflearn.fully_connected(net, 8, activation='tanh',  weights_init=tnorm)    
    net = tflearn.fully_connected(net, 1 , activation='tanh', weights_init=tnorm)
    net = tflearn.regression(net, optimizer='momentum', loss='mean_square', learning_rate=0.001)

    model = tflearn.DNN(net, tensorboard_verbose=0, clip_gradients=1.0)
    model.fit(X_train, Y_train, n_epoch = 10, validation_set=0.1, show_metric=True, run_id="Regression")
`

And here's the output to the network:
`Training Step: 600  | total loss: 0.010
| Momentum | epoch: 003 | loss: 0.010 - acc: 1.0000 -- iter 10000/10000: 
`
